packages/expo-modules-core/src/ts-declarations/global.ts

```diff
@@ -0,0 +1,41 @@
+import type { EventEmitter } from './EventEmitter';
+import type { SharedObject } from './SharedObject';
+
+/* eslint-disable no-var */
+
+declare global {
+  /**
+   * Global object containing all the native bindings installed by Expo.
+   * The object may not be available if Expo modules are not set up correctly in the project.
+   */
+  var expo: {
+    /**
+     * Host object that is used to access native Expo modules.
+     */
+    modules: Record<string, any>;
+
+    // Natively defined JS classes
+
+    /**
+     * @see EventEmitter
+     */
+    EventEmitter: typeof EventEmitter;
+
+    /**
+     * @see SharedObject
+     */
+    SharedObject: typeof SharedObject;
+
+    // Utils
+
+    /**
+     * Generates a random UUID v4 string.
+     */
+    uuidv4(): string;
+
+    /**
+     * Generates a UUID v5 string representation of the value in the specified namespace.
+     */
+    uuidv5(name: string, namespace: string): string;
+  };
```

Comment:
> The object may not be available if Expo modules are not set up correctly in the project.

Nit, according to the comment above we might want to reflect that case also in global type.

```suggestion
  } | undefined;
```

Reply:
I think it would be better to keep it non-optional. The declaration is visible only when you already have expo package installed and in most cases the object will be set properly. RN autolinking is enough for this object to be set.
Right now it is `undefined` on Web, but we're going to change it pretty soon.